### PR TITLE
Feature/mds 2652 generate documents with context

### PR DESCRIPTION
**Pull request by Jsyro**

### CODE REVIEW

File: migrations/sql/afterMigrate__1__seed_data.sql [hunk 1]
```diff
@@ -791,7 +799,9 @@ VALUES
       "label": "Property",
       "type": "FIELD",
       "placeholder": "Enter the property",
-      "required": true
+      "required": true,
+      "rel_data_path": "now_application.property_name",
+      "read-only": true
```
Comment (pdittaro) [inline migrations/sql/afterMigrate__1__seed_data.sql]: Should be consistent underscores.

File: migrations/sql/V2020.02.25.14.29__document_object_traversal_data.sql [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    {
+      "id": "inspector",
+      "label": "Inspector",
+      "type": "FIELD",
+      "placeholder": "Enter the inspector''s name",
+      "required": true,
+      "rel_data_path": "now_application.lead_inspector.name"
+    }
+  ]'
+where document_template_code = 'NOW-CAL';
```
Comment (nayr974) [inline migrations/sql/V2020.02.25.14.29__document_object_traversal_data.sql]: Please update these type codes

File: migrations/sql/V2020.02.25.14.29__document_object_traversal_data.sql [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    {
+      "id": "inspector",
+      "label": "Inspector",
+      "type": "FIELD",
+      "placeholder": "Enter the inspector''s name",
+      "required": true,
+      "rel_data_path": "now_application.lead_inspector.name"
+    }
+  ]'
+where document_template_code = 'NOW-WDL';
```
Comment (nayr974) [inline migrations/sql/V2020.02.25.14.29__document_object_traversal_data.sql]: Please update these type codes

File: services/core-api/app/api/document_generation/models/document_template.py [hunk 4]
```diff
@@ -1,23 +1,68 @@
...
+            if not relative_data_path:
+                current_app.logger.debug(f'No rel_data_path for {item["id"]}')
+                continue
+
+            current_object = source_obj_instance
+            for x in relative_data_path.split('.'):
+                current_app.logger.debug(f'getting {current_object}.{x}')
+                current_object = getattr(current_object, x)
+
+            current_app.logger.info(
```
Comment (pdittaro) [inline services/core-api/app/api/document_generation/models/document_template.py]: I think this is too much info still for an 'info' logging. Info should probably just give information about the form being created, doesn't need the field level information.

File: services/core-api/app/api/now_applications/resources/now_application_document_type_resource.py [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import uuid
+from flask import request, current_app
```
Comment (pdittaro) [inline services/core-api/app/api/now_applications/resources/now_application_document_type_resource.py]: Are you using request in this file?

File: services/docgen-api/index.js [hunk 6]
```diff
@@ -8,7 +8,7 @@ const express = require(`express`);
 const bodyParser = require(`body-parser`);
 const crypto = require("crypto");
 
-const templatedir = "/tmp/tmp-reports/";
+const templatedir = "/app/templates/";
```
Comment (pdittaro) [inline services/docgen-api/index.js]: I don't think this is correct, was updated to /tmp/tmp-reports in the other PR

File: migrations/sql/afterMigrate__1__seed_data.sql (hunk 1 above)
Comment (Jsyro) [reply migrations/sql/afterMigrate__1__seed_data.sql]: all '-'s

File: services/core-api/app/api/document_generation/models/document_template.py (hunk 4 above)
Comment (Jsyro) [reply services/core-api/app/api/document_generation/models/document_template.py]: agreed

File: services/docgen-api/index.js (hunk 6 above)
Comment (pdittaro) [reply services/docgen-api/index.js]: Nevermind, see that you changed.

File: services/core-api/app/api/now_applications/resources/now_application_document_type_resource.py (hunk 5 above)
Comment (Jsyro) [reply services/core-api/app/api/now_applications/resources/now_application_document_type_resource.py]: Line 31.

File: services/docgen-api/index.js (hunk 6 above)
Comment (Jsyro) [reply services/docgen-api/index.js]: fixed.

File: services/core-web/src/reducers/rootReducer.js [hunk 7]
```diff
@@ -63,6 +65,7 @@ export const reducerObject = {
   ...reportReducer,
   ...userReducer,
   ...noticeOfWorkReducer,
+  [DOCUMENTS]: documentReducer,
```
Comment (nayr974) [inline services/core-web/src/reducers/rootReducer.js]: IMO please either keep this consistent with the other reducer types, or change the others to match this.

Comment (Jsyro) [reply services/core-web/src/reducers/rootReducer.js]: Done

**Comment by Jsyro** [comment]
closeing to reseed data.

**Pull request merged**